### PR TITLE
fix(client): stop margin from pushing outline out of the container

### PR DESCRIPTION
**Pull request by Sboonny**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Help camper with this PR #49579

because logo CSS is pushing the `outline` outside its container while its container is smaller than the parent container, so there is no outline at the bottom and top of the logo

The final desired look.
image

### CODE REVIEW

**Comment by Sboonny** [comment]
the goal of the PR to have this focus outline, but because it is changing things, which I needed to change, it won't hurt to check the layout in mobile and laptop as well.  

image

File: client/src/components/Header/components/universal-nav.css [hunk 1]
```diff
@@ -574,11 +556,13 @@ button.nav-link[aria-disabled='true'] {
   .toggle-button-nav {
     padding: 2px 8px;
   }
+  #universal-nav-logo {
+    padding-top: 0.35em;
+  }
   #universal-nav-logo svg {
     display: block;
     width: auto;
     height: 18px;
```
Comment (Sboonny) [inline client/src/components/Header/components/universal-nav.css:565]:
```suggestion
```

File: client/src/components/Header/components/universal-nav.css [hunk 2]
```diff
@@ -574,11 +556,9 @@ button.nav-link[aria-disabled='true'] {
...
   }
-  #universal-nav-logo svg {
-    display: block;
-    width: auto;
-    height: 18px;
-    margin: 10px 0;
+  #universal-nav-logo {
+    padding-top: 0.35em;
+  }
   }
```
Comment (Sboonny) [inline client/src/components/Header/components/universal-nav.css:562]:
```suggestion
```

File: client/src/components/Header/components/universal-nav.css [hunk 3]
```diff
@@ -25,15 +25,12 @@
...
 
-.universal-nav-middle {
+#universal-nav-logo {
   display: flex;
   align-items: center;
-  flex: 1 0 33%;
-  overflow-x: auto;
-  overflow-y: hidden;
+  overflow: hidden;
+  max-inline-size: max-content;
```
Comment (Sboonny) [inline client/src/components/Header/components/universal-nav.css:32]: Flex and grid have a different way of calculating the width and height of SVG, for different browsers.

this aims to stop that behavior

**Comment by Sboonny** [comment]
Just a heads up, I know this will break RTL, just don't want to add RTL fix until this is approved

**Review by moT01 — COMMENTED** [review]
Looks mostly good to me @Sboonny 👍

The logo seems slight off of vertical center on mobile:
Screen Shot 2023-03-13 at 3 08 55 PM

^ A slight bit low maybe? Looks a lot better than it did though - so I would be fine with it like this. If you feel like giving it a quick fix, feel free.

**Comment by Sboonny** [comment]
Whelp, you have found another 1px bug 

[rec-tab (18).webm](https://user-images.githubusercontent.com/88248797/224932168-bedea1df-c2f0-4e7d-8262-04af7f47de9b.webm)

**Review by moT01 — APPROVED** [review]
I'm not gonna lie, it looks 1px too high 😄
Screen Shot 2023-03-14 at 10 01 54 AM

I would be fine with this whenever.

**Review by moT01 — APPROVED** [review]

**Review by raisedadead — APPROVED** [review]

**Pull request merged**